addon/src/-internal/build-registry.ts

```diff
@@ -141,20 +145,27 @@ export default function buildRegistry(resolver: Resolver) {
   }) as unknown as Owner;
 
   // @ts-ignore: this is private API.
-  let container = registry.container({ owner: owner });
+  const container = registry.container({ owner: owner });
   // @ts-ignore: this is private API.
   owner.__container__ = container;
 
   exposeRegistryMethodsWithoutDeprecations(container);
 
-  if (has('ember-data/setup-container')) {
-    // ember-data is a proper ember-cli addon since 2.3; if no 'import
-    // 'ember-data'' is present somewhere in the tests, there is also no `DS`
-    // available on the globalContext and hence ember-data wouldn't be setup
-    // correctly for the tests; that's why we import and call setupContainer
-    // here; also see https://github.com/emberjs/data/issues/4071 for context
-    let setupContainer = require('ember-data/setup-container')['default'];
-    setupContainer(owner);
+  if (macroCondition(dependencySatisfies('ember-data', '>= 2.3'))) {
```

Comment:
This change is also meaningful -- we don't want to use requirejs -- we're trying to get rid of it.

Insetad, our abstraction over runtime module loader is provided by embroider-macros.

Additionally, I saw that in this file that setup-container is deprecated without replacement for removal in v6.

So I added another if condition around this code so we can hopefully be a little fault tolerant when ember-data v6 comes around.

Reply:
This could also be pulled out to a separate PR, if folks want